docs/relational-databases/performance/best-practice-with-the-query-store.md

```diff
@@ -425,6 +425,55 @@ Consider the following options:
 > [!TIP]
 > When using an Object-Relational Mapping (ORM) solution such as Entity Framework (EF), application queries like manual LINQ query trees or certain raw SQL queries may not be parameterized, which impacts plan re-use and the ability to track queries in the Query Store. For more information, see [EF Query caching and parameterization](/ef/core/performance/advanced-performance-topics) and [EF Raw SQL Queries](/ef/core/querying/raw-sql).
 
+### Find non-parameterized queries in Query Store
+
+You can find the number of plans stored in Query Store using the below query, using query store DMVs:
+
+```sql
+  SELECT count(plan_id) plan_count, query_hash,Txt.query_text_id, Txt.query_sql_text
+  FROM sys.query_store_plan AS Pl
+  INNER JOIN sys.query_store_query AS Qry
+      ON Pl.query_id = Qry.query_id
+  INNER JOIN sys.query_store_query_text AS Txt
+      ON Qry.query_text_id = Txt.query_text_id
+         group by query_hash,Txt.query_text_id, Txt.query_sql_text
+         order by 1 desc
+
+```
+
+On SQL Server, this extended event session creates an event file in the SQL Server Log folder by default. For example, in a default SQL Server 2019 installation on Windows, the event file (.xel file) should be created in the folder `C:\Program Files\Microsoft SQL Server\MSSQL15.MSSQLSERVER\MSSQL\Log`. For Azure SQL Managed Instance, specify an Azure Blob Storage location instead. For more information, see [XEvent event_file for Azure SQL Managed Instance](/azure/azure-sql/database/xevent-code-event-file#phase-2-transact-sql-code-that-uses-azure-storage-container). The event 'qds.query_store_db_diagnostics' is not available for Azure SQL Database.  
+
+The following sample creates a session to capture the event `query_store_db_diagnostics`:
+
+```sql
+  CREATE EVENT SESSION [QueryStore_Troubleshoot] ON SERVER 
+  ADD EVENT qds.query_store_db_diagnostics(
+      ACTION(sqlos.system_thread_id,sqlos.task_address,sqlos.task_time,sqlserver.database_id,sqlserver.database_name))
+  ADD TARGET package0.event_file(SET filename=N'QueryStore',max_file_size=(100))
+  WITH (MAX_MEMORY=4096 KB,EVENT_RETENTION_MODE=ALLOW_SINGLE_EVENT_LOSS,MAX_DISPATCH_LATENCY=30 SECONDS,MAX_EVENT_SIZE=0 KB,MEMORY_PARTITION_MODE=NONE,TRACK_CAUSALITY=OFF,STARTUP_STATE=OFF)
+```
+
+With this data you can find plan count in the Query Store, and also many other stats as well. Look for the `plan_count`, `query_count`, `max_stmt_hash_map_size_kb`, and `max_size_mb` columns in the event data, in order to understand the amount of memory used and number of plans that are tracked by Query Store. If the plan_count is higher than normal, it may indicate an increase in non-parameterized queries. Use the below Query Store DMVs query to review the parameterized queries and non-parameterized queries in the Query Store.
```

Comment:
```suggestion
With this data you can find plan count in the Query Store, and also many other stats as well. Look for the `plan_count`, `query_count`, `max_stmt_hash_map_size_kb`, and `max_size_mb` columns in the event data, in order to understand the amount of memory used and number of plans that are tracked by Query Store. If the plan count is higher than normal, it may indicate an increase in non-parameterized queries. Use the below Query Store DMVs query to review the parameterized queries and non-parameterized queries in the Query Store.
```

Reply:
@WilliamDAssafMSFT, Good catch, Applied your suggested changes. Thanks!!